evalml/model_understanding/_partial_dependence_fast_mode_utils.py

```diff
@@ -0,0 +1,100 @@
+import pandas as pd
+
+from evalml.utils.woodwork_utils import infer_feature_types
+
+
+def _get_cloned_feature_pipelines(
+    features,
+    pipeline,
+    variable_has_features_passed_to_estimator,
+    X_training,
+    y_training,
+):
+    """Clones and fits pipelines for partial dependence fast mode.
+
+    Args:
+        pipeline (PipelineBase or subclass): Fitted pipeline that will be cloned.
+        features (string, tuple[int or string]): The target feature for which to create the partial dependence plot for.
+            If features is a string, it must be a valid column name in X.
+            If features is a tuple of int/strings, it must contain valid column integers/names in X.
+        X_training (pd.DataFrame, np.ndarray): The data that was used to train the original pipeline. Will
+            be used to train the cloned pipelines.
+        y_training (pd.Series, np.ndarray): The target data that was used to train the original pipeline. Will
+            be used to train the cloned pipelines.
+
+    Returns:
+        dict[str, PipelineBase or subclass]: Dictionary mapping feature name to the pipeline pipeline
+            fit for it.
+    """
+    if X_training is None or y_training is None:
+        raise ValueError("Training data is required for partial dependence fast mode.")
+
+    X_training = infer_feature_types(X_training)
+
+    # Make sure that only components that are capable of handling fast mode are in the pipeline
+    new_parameters = pipeline.parameters
+    for component in pipeline.component_graph.component_instances.values():
+        new_parameters = component._handle_partial_dependence_fast_mode(
+            X_training,
+            new_parameters,
+        )
+
+    # Create a fit pipeline for each feature
+    cloned_feature_pipelines = {}
+    for variable in features:
+        # Don't fit pipelines if the feature has no impact on predictions
+        if not variable_has_features_passed_to_estimator[variable]:
```

Comment:
can you give me an example of when this happens in practice? Can't think of one off the top of my head!

Reply:
If the `RFRegressorSelectFromModel` decides that the same feature you're trying to calculate PD for is unimportant and should be dropped prior to passing the data to the estimator. 

It's what lets us just use the original data's average predictions as PD values. We know we won't have to change that feature's values and re-transform, so there's no need to waste time fitting a cloned pipeline for that feature!

Reply:
gotcha - I think I understand how everything is linked now. Thanks!